src/Raku/ast/type.rakumod

```diff
@@ -474,8 +474,22 @@ class RakuAST::Type::Parameterized
             my $args := $!args.IMPL-COMPILE-TIME-VALUES;
             my @pos := $args[0];
             my %named := $args[1];
-            my $ptype := self.IMPL-BASE-TYPE.compile-time-value;
-            $ptype.HOW.parameterize($ptype, |@pos, |%named)
+            my $ptype := self.base-type.compile-time-value;
+            my $phow  := $ptype.HOW;
+
+            # There are cases where the type is actually incapable of
+            # parameterization, at which point we pass it back unmodified.
+            if nqp::istype($phow, Perl6::Metamodel::ClassHOW) {
```

Comment:
Can't this whole if/else be simplified to just the else block?

Reply:
I have it arranged this way due to experimentation / trial and error, but yes, you are correct.

Note also that this commit is not performing as advertised due to a mess up with my environment variables during testing.